apps/condo/domains/common/utils/text.js

```diff
@@ -0,0 +1,24 @@
+function normalizeText (text) {
+    if (!text) return
+    String(text).normalize()
+    const punctuations = '.,:'
+    return text
+        // replace three or more EOL to two EOL
+        .replace(/\n{3,}/gm, '\n\n')
```

Comment:
```suggestion
        .replace(/\n+/gm, '\n')
```

Reply:
Ok, it seems, like it's better to disallow blank lines at all.